examples/job-catalog/src/openai.ts

```diff
@@ -74,6 +127,38 @@ client.defineJob({
       model: "text-embedding-ada-002",
       input: "The food was delicious and the waiter...",
     });
+
+    //Function call example for OpenAI
+    const createCompletion = async () => {
+      const chatCompletion = await io.openai.createChatCompletion("chat-completion", {
+        model: "gpt-3.5-turbo",
+        messages: chatMessages,
+        functions,
+        function_call: "auto"
+      });
+      return chatCompletion
+    }
+
+    const chatCompletion = await createCompletion();
+
+    // @ts-ignore
+    const response = chatCompletion?.data?.choices[0].message;
+
+    if ('function_call' in response) {
+      const function_name: string = "get_current_affairs";
+      const function_call = available_functions[function_name];
+      const args = JSON.parse(response['function_call']['arguments'])
+      // @ts-ignore
+      const fn_response = await function_call(...Object.values(args));
+      chatMessages.push({
+        role: 'function',
+        name: function_name,
+        content: fn_response
+      })
+      const finalCompletion = await createCompletion();
+      // @ts-ignore
+      await io.logger.info(finalCompletion?.data?.choices[0].message);
```

Comment:
In general, could you add more comments in the code to clearly explain what is going on?

Reply:
Added more comments now !!